daemon/daemon.go

```diff
@@ -359,7 +360,22 @@ func (daemon *Daemon) restore() error {
 			if daemon.configStore.AutoRestart && container.shouldRestart() {
 				logrus.Debugf("Starting container %s", container.ID)
 
-				if err := daemon.containerStart(container); err != nil {
+				var retries int
+				for err = daemon.containerStart(container); err != nil && retries <= 5; retries++ {
+					// Best effort to wait for container deps to be up
+					switch e := err.(type) {
+					case errcode.Error:
+						if e.ErrorCode() != derr.ErrorCodeLinkNotRunning {
+							break
+						}
+						err = nil
```

Comment:
I can make this not a switch for now to clean this up, but I know we have another case where this is an issue as well: #18091